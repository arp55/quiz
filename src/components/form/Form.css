.header-typo {
  margin-bottom: 10px;
}
