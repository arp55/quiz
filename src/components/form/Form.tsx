import { MenuItem, Select, FormControl, Paper, TextField, Typography, FormLabel, RadioGroup, FormControlLabel, Radio, InputLabel, Button } from '@mui/material'
import React, { useState } from 'react'
import { NavigateFunction } from 'react-router-dom'
import './Form.css'

type Props = {
    navigate: NavigateFunction
}

export default function Form({ navigate }: Props) {
    const [val, setval] = useState({
        name: '',
        gender: '',
        age: '',
        language: ''
    })

    function handleChange(value: string, e: any) {
        setval({ ...val, [value]: e.target.value })
    }

    function handleSubmit() {
        // e.preventDefault()`
        if (val.name.length !== 0) {
            navigate('quiz')
        }
    }

    return (
        <Paper style={{ maxWidth: '400px', padding: '20px' }}>
            {/* <form onSubmit={handleSubmit}> */}
            <Typography data-testid='text-header' className='header-typo' variant='h6'>Please fill the details</Typography><br />
            <FormControl>
                <TextField data-testid='form-input' inputProps={{ 'data-testid': 'name-input' }} label="Name" variant="standard" placeholder='Please enter name' onChange={(e) => handleChange('name', e)} />
                <FormLabel id="radio-gender">Gender</FormLabel>
                <RadioGroup
                    aria-labelledby="radio-gender"
                    // defaultValue=""
                    value={val.gender}
                    name="radio-group"
                    onChange={(e) => handleChange('gender', e)}
                >
                    <FormControlLabel value="female" control={<Radio />} label="Female" />
                    <FormControlLabel value="male" control={<Radio />} label="Male" />
                </RadioGroup>
                <FormControl>
                    <InputLabel id="select-label">Age</InputLabel>
                    <Select
                        data-testid="age-input"
                        labelId="select-label"
                        id="select"
                        value={val.age}
                        label="Age"
                        onChange={(e) => handleChange('age', e)}
                    >
                        <MenuItem value={10}>10</MenuItem>
                        <MenuItem value={20}>20</MenuItem>
                        <MenuItem value={30}>30</MenuItem>
                    </Select>

                </FormControl>
                <FormLabel id="radio-language">Test language</FormLabel>
                <RadioGroup
                    data-testid='language-radio'
                    aria-labelledby="radio-language"
                    // defaultValue=""
                    value={val.language}
                    name="radio-group"
                    onChange={(e) => handleChange('language', e)}
                >
                    <FormControlLabel value="spanish" control={<Radio />} label="spanish" />
                    <FormControlLabel value="english" control={<Radio />} label="english" />
                    <FormControlLabel value="hindi" control={<Radio />} label="hindi" />
                </RadioGroup>
            </FormControl>
            <div style={{ width: '100%', textAlign: 'center' }}>
                <Button data-testid='submit-btn' disabled={val.name.length === 0} variant="outlined" onClick={handleSubmit} >Submit</Button>
            </div>
            {/* </form> */}
        </Paper >
    )
}
