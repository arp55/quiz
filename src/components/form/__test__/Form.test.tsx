import { cleanup, render, fireEvent } from '@testing-library/react'
import React from 'react'
import renderer from 'react-test-renderer'
import Form from '../Form'

afterEach(cleanup)

describe('test for form', () => {
    test('snapshot for form comp', () => {
        const tree = renderer.create(<Form navigate={() => { }} />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('header present in form', () => {
        const { getByTestId } = render(<Form navigate={() => { }} />)
        expect(getByTestId('text-header')).toBeInTheDocument()
        expect(getByTestId('text-header')).toHaveTextContent('Please fill the details')
    })

    test('textinput present in form', () => {
        const { getByTestId } = render(<Form navigate={() => { }} />)
        expect(getByTestId('form-input')).toHaveTextContent('Name');
    })

    test('submit button to be disabled by default', () => {
        const { getByTestId, getByText } = render(<Form navigate={() => { }} />)
        expect(getByTestId('name-input')).toBeInTheDocument()
        expect(getByText('Submit')).toBeInTheDocument()
        expect(getByText('Submit')).toBeDisabled()
    })

    test('submit button to be enabled', () => {
        const { getByTestId, getByText } = render(<Form navigate={() => { }} />)
        const input = getByTestId('name-input')
        fireEvent.change(input, { target: { value: 'rohit' } })
        expect(getByText('Submit')).toBeInTheDocument()
        expect(getByText('Submit')).toBeEnabled()
    })

    test('language radio buttons to be present', () => {
        const { getByTestId } = render(<Form navigate={() => { }} />)
        expect(getByTestId('language-radio')).toBeInTheDocument()
    })

    test('age select buttons to be present', () => {
        const { getByTestId } = render(<Form navigate={() => { }} />)
        expect(getByTestId('age-input')).toBeInTheDocument()
    })
})