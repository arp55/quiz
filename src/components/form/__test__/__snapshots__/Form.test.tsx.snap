// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for form snapshot for form comp 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
  style={
    Object {
      "maxWidth": "400px",
      "padding": "20px",
    }
  }
>
  <h6
    className="MuiTypography-root MuiTypography-h6 header-typo css-2ulfj5-MuiTypography-root"
    data-testid="text-header"
  >
    Please fill the details
  </h6>
  <br />
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      data-testid="form-input"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Name
      </label>
      <div
        className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          data-testid="name-input"
          disabled={false}
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Please enter name"
          required={false}
          type="text"
        />
      </div>
    </div>
    <label
      className="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
      id="radio-gender"
    >
      Gender
    </label>
    <div
      aria-labelledby="radio-gender"
      className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
      role="radiogroup"
    >
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            disabled={false}
            name="radio-group"
            onChange={[Function]}
            type="radio"
            value="female"
          />
          <span
            className="css-hyxlzm"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Female
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            disabled={false}
            name="radio-group"
            onChange={[Function]}
            type="radio"
            value="male"
          />
          <span
            className="css-hyxlzm"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Male
        </span>
      </label>
    </div>
    <div
      className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        id="select-label"
      >
        Age
      </label>
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        data-testid="age-input"
        onClick={[Function]}
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="select-label select"
          className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="select"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="notranslate"
          >
            â€‹
          </span>
        </div>
        <input
          aria-hidden={true}
          className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          disabled={false}
          onAnimationStart={[Function]}
          onChange={[Function]}
          required={false}
          tabIndex={-1}
          value=""
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1ftyaf0"
          >
            <span>
              Age
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <label
      className="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
      id="radio-language"
    >
      Test language
    </label>
    <div
      aria-labelledby="radio-language"
      className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
      data-testid="language-radio"
      role="radiogroup"
    >
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            disabled={false}
            name="radio-group"
            onChange={[Function]}
            type="radio"
            value="spanish"
          />
          <span
            className="css-hyxlzm"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          spanish
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            disabled={false}
            name="radio-group"
            onChange={[Function]}
            type="radio"
            value="english"
          />
          <span
            className="css-hyxlzm"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          english
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            disabled={false}
            name="radio-group"
            onChange={[Function]}
            type="radio"
            value="hindi"
          />
          <span
            className="css-hyxlzm"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          hindi
        </span>
      </label>
    </div>
  </div>
  <div
    style={
      Object {
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <button
      className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      data-testid="submit-btn"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      Submit
    </button>
  </div>
</div>
`;
